package/CHANGELOG

```
@@ -34,6 +34,10 @@ Fixes
     (e.g. bonds, angles) (PR #3779).
 
 Enhancements
+  * Improve C content of libxdr Cython, add `read_direct` methods to read 
+    coordinates, velocities and forces directly into memoryviews of `Timestep`
+    attributes, make `TRR` timestep have positions, velocities and forces on
+    `__init__`. (Issue #3891 PR  #3892).
```

add under Fixes: XTC and TRR readers now fail with IOError when a status except EOK (=0) is detected on reading a frame instead of silently continuing

---

Some readers had enough redundancy built in to avoid not continuing with  `retcode != EOK` but at least its now done consistently.

---

This was the pattern that protected some branches 

```python
        if return_code != EOK and return_code != EENDOFFILE \
           and return_code != EINTEGER:
            raise IOError('TRR read error = {}'.format(
                **error_message[return_code]))**

        if return_code == EENDOFFILE or return_code == EINTEGER:
            self.reached_eof = True
            raise StopIteration

# can only be EOK at this point but it was a bit opaque. 

```

---

Ok, good to know that we probably do not change a lot of behavior. Nevertheless, the current approach is cleaner.

(Btw, while somewhere I came across my own explanation of why TRR stops with EINTEGER and I think it was related to not being able to read the magic number.)